gtsfm/common/postgres_client.py

```diff
@@ -0,0 +1,243 @@
+        """
+        Check if a table exists in the database
+        
+        Args:
+            table_name (str): Name of the table to check
+            
+        Returns:
+            bool: True if table exists, False otherwise
+        """
+        query = """
+        SELECT EXISTS (
+            SELECT FROM information_schema.tables 
+            WHERE table_schema = 'public' 
+            AND table_name = %s
+        );
+        """
+        result = self.fetch_one(query, (table_name,))
+        return result[0] if result else False
+    
+    def create_table_if_not_exists(self, table_name: str, create_query: str) -> bool:
+        """
+        Create a table if it doesn't exist
+        
+        Args:
+            table_name (str): Name of the table
+            create_query (str): SQL CREATE TABLE statement
+            
+        Returns:
+            bool: True if successful
+        """
+        if not self.table_exists(table_name):
+            return self.execute(create_query)
+        return True
+    
+    def __getstate__(self) -> Dict[str, Any]:
+        """Custom serialization to avoid serializing connection objects"""
+        state = self.__dict__.copy()
+        # Do not serialize connection and cursor
+        state['conn'] = None
+        state['cursor'] = None
+        return state
+    
+    def __setstate__(self, state: Dict[str, Any]) -> None:
+        """Custom deserialization"""
+        self.__dict__.update(state)
+        # Connection will be re-established when needed
+    
+    def ensure_schema(self) -> bool:
+        """Ensure database schema is initialized"""
+        if not self._schema_initialized:
+            self._schema_initialized = self.initialize_gtsfm_schema()
+        return self._schema_initialized
+    
+    def execute_with_schema_check(self, query: str, params: Optional[Tuple] = None) -> bool:
+        """Execute query after ensuring schema exists"""
+        if not self.ensure_schema():
+            logger.error("Failed to initialize database schema")
+            return False
+        return self.execute(query, params)
+    
+    def initialize_gtsfm_schema(self) -> bool:
+        """Initialize all GTSFM-related database tables"""
+        try:
+            # Create two-view results table
+            if not self.execute(self._get_two_view_results_table_ddl()):
+                logger.error("Failed to create two_view_results table")
+                return False
+            
+            # Create two-view reports table  
+            if not self.execute(self._get_two_view_reports_table_ddl()):
+                logger.error("Failed to create two_view_reports table")
+                return False
+            
+            logger.info("Database schema initialized successfully")
+            return True
+        except Exception as e:
+            logger.error(f"Failed to initialize database schema: {e}")
+            return False
+    
+    def _get_two_view_results_table_ddl(self) -> str:
```

Comment:
this logic should ideally move to the two view estimation class. this class should contain only "common" postgres client code.